esp-hal/src/uart.rs

```diff
@@ -510,27 +463,76 @@ where
     }
 }
 
+impl<'d, T> UartRx<'d, T, Blocking>
+where
+    T: Instance + 'd,
+{
+    /// Create a new UART RX instance in [`Blocking`] mode.
+    pub fn new<RX: InputPin>(
+        _uart: impl Peripheral<P = T> + 'd,
+        rx: impl Peripheral<P = RX> + 'd,
+    ) -> Self {
+        crate::into_ref!(rx);
+        rx.set_to_input(Internal);
+        rx.connect_input_to_peripheral(T::rx_signal(), Internal);
+        Self::new_inner()
```

Comment:
When creating either side from the raw peripheral, we need to do the same initialization that we do when creating `Uart` like setting up the baud rate etc, which is probably why the test is failing.

Reply:
That means we should also accept a `Config` in the constructor too